src/cpp/session/modules/SessionFind.cpp

```diff
@@ -270,29 +270,32 @@ class GrepOperation : public boost::enable_shared_from_this<GrepOperation>
 
       std::string::iterator inputPos = pContent->begin();
 
+      std::size_t nUtf8CharactersProcessed = 0;
       smatch match;
       while (regex_search(std::string(inputPos, pContent->end()), match,
                           regex("\x1B\\[(\\d\\d)?m(\x1B\\[K)?")))
       {
-         std::string match1 = match[1];
-
-         decodedLine.append(decode(
-               std::string(inputPos, inputPos + match.position())));
-
+         // decode the current match, and append it
+         std::string decoded = decode(std::string(inputPos, inputPos + match.position()));
+         decodedLine.append(decoded);
          inputPos += match.position() + match.length();
-
-         size_t charSize;
-         Error error = string_utils::utf8Distance(decodedLine.begin(),
-                                                  decodedLine.end(),
+         
+         // count the number of UTF-8 characters processed
+         std::size_t charSize;
+         Error error = string_utils::utf8Distance(decoded.begin(),
```

Comment:
One potential issue -- if the data is streamed in byte by byte, is it possible that we get 'half' of a UTF-8 character upon processing?

If so, we might want to keep the logic here as-is but limit the size we search.